Updated docs for ROCm 7.x.x (#239)
ROCm-Developer-Tools/ROC_SHMEM: CHANGELOG.md

# Changelog for rocSHMEM

## Unreleased - rocSHMEM 3.x.x for ROCm 7.x.x
### Added
* Allow for IPC, RO, GDA backends to be selected at runtime
* Added the GDA conduit for different NIC vendors
   * AMD Pensando IONIC
   * Broadcom BNXT\_RE (Thor 2)
   * Mellanox MLX5 (IB and RoCE ConnectX-7)
* Added new APIs:
   * `rocshmem_get_device_ctx`

### Changed
* The following APIs have been deprecated:
  * `rocshmem_wg_init`
  * `rocshmem_wg_finalize`
  * `rocshmem_wg_init_thread`

> * `rocshmem_ptr`  has an implementation that can return non-null pointer to

Reviewer: ```suggestion
* `rocshmem_ptr`  can now return non-null pointer to
```

Would it make sense to simplify the wording as above?

   a shared memory region when the IPC transport is available to reach that region.
   Previously, it would return a null pointer.
* `ROCSHMEM_RO_DISABLE_IPC` was renamed to `ROCSHMEM_DISABLE_MIXED_IPC`.

>   This enviroment variable was not documented for prior releasees.

Reviewer: ```suggestion
  This enviroment variable was not documented for prior releases.
```

>   However, we want to inform users who were using this undocumented feature.

Reviewer: ```suggestion
  It is now documented to inform users who were using this undocumented feature.
```

# Changelog for rocSHMEM

## Unreleased - rocSHMEM 3.x.x for ROCm 7.x.x

### Added

* Added the GDA conduit for different NIC vendors

>    * Mellanox ConnectX-7 (IB and RoCE)

Reviewer 2: It is expected that CX6 would also work, even if untested, maybe we should refer to it as MLX5.
```suggestion
   * Mellanox MLX5 (IB and RoCE ConnectX-7)
```

Author: That works for me, I'll adjust Broadcom Thor 2 to refer to bnxt_re to follow the pattern

   * Broadcom Thor 2
* Added new APIs:
   * `rocshmem_get_device_ctx`

>    * `rocshmem_ptr`

Reviewer 2: rocshmem_ptr was already existing in prior releases, but would always return nullptr (a valid implementation of the standard).

### Changed

* `rocshmem_ptr` has an implementation that can return non-null pointer to a shared memory region when the IPC transport is available to reach that region.
